vars/abapEnvironmentPipelineStageInit.groovy

```diff
@@ -0,0 +1,54 @@
+import groovy.transform.Field
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.ConfigurationLoader
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = []
+@Field STAGE_STEP_KEYS = []
+@Field Set STEP_CONFIG_KEYS = []
+@Field Set PARAMETER_KEYS = []
+/**
+ * This stage initializes the ABAP Environment Pipeline run
+ */
+void call(Map parameters = [:]) {
+
+
+    def script = checkScript(this, parameters) ?: this
+
+    def stageName = parameters.stageName?:env.STAGE_NAME
+
+    piperStageWrapper (script: script, stageName: stageName, stashContent: [], ordinal: 1, telemetryDisabled: true) {
+
+        //deleteDir()
+        checkout scm
+
+        setupCommonPipelineEnvironment script: script
```

Comment:
please make sure to consider custom defaults.
You can find an example here:
https://github.com/SAP/jenkins-library/blob/master/vars/piperPipelineStageInit.groovy#L52